clients/cobol-lsp-vscode-extension/src/services/reporter/TelemetryReporterImpl.ts

```diff
@@ -51,14 +51,17 @@ export class TelemetryReporterImpl implements TelemetryReport {
     };
   }
 
-  private reporter: TelemetryReporter;
+  private reporter: TelemetryReporter | undefined;
 
-  constructor(private telemetryKeyId: string) {
-    this.reporter = new TelemetryReporter(this.telemetryKeyId);
+  constructor(private telemetryKeyId: string | undefined) {
```

Comment:
I don't think this is ever undefined.
```suggestion
  constructor(private telemetryKeyId: string) {
```